### PR TITLE
APPS-1017 Bug: stabilize prestashop

### PR DESCRIPTION
**Pull request by multinauta**
Refer:
 - https://shipitchile.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=APPS&modal=detail&selectedIssue=APPS-1050
 - https://shipitchile.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=APPS&modal=detail&selectedIssue=APPS-1051
 - https://shipitchile.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=APPS&modal=detail&selectedIssue=APPS-1052
 - https://shipitchile.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=APPS&modal=detail&selectedIssue=APPS-1056

### Issues founds and solutions

error

**Bugs**

**Problem**:  as of version 1.6.1.0, the carriers table began to create a new record, and not modifying the existing one, causing the reference id to be read incorrectly.

**Solution**: i created an intermediate query to get the correct carrier id

--

**Problem**: the module was incorrectly reading the request that returned "integrations/seller/prestashop"

**Solution**: i modified the code so that it read correctly from the corresponding node

--

**Problem**: Undefined Offset,  uninitialized variable items.

**Solutions**: i initialized the variable

---

**Improvement**

- Register the secret key for the webhook

- Standardize the statuses and add translations

- I added log to the objects sent in each service for debugging if necessary

- Created the clases Shipment, ShipitDestiny, ShipitSize, ShipitInsurance, ShipitProduct and ShipitCourier

- I built the shipment object using the created classes

- I added 3 methods to the Tools class: 
   **getClientName():** get the courier's name.
   **getCourierId:** get the courier id through the name from the courier endpoint in prices
   **splitAddressAndNumber():** separate name and number from address

- I modified the processWebhook function to be able to update the shipment status

---

**Staging Test**

staging

**Sales**

app

**Shipment**

Screenshot from 2021-01-27 21-34-40

I extended the configuration view to show the error log

image

### CODE REVIEW

File: rg_shipit/classes/ShipitCityTrack.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    function getCanceled(){
+      return $this->canceled;
+    }
+
+    function getArchived(){
+      return $this->archived;
+    }
+  }
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitCityTrack.php]: Missing Empty Line at EOF

File: rg_shipit/views/templates/admin/configure-bottom.tpl [hunk 2]
```diff
@@ -30,3 +37,13 @@
         </ul>
     </p>
 </div>
+
+<div class="panel">
+    <h3><i class="icon icon-tags"></i> Log de Errores</h3>
+    <p>
+
```
Comment (mgraterol) [inline rg_shipit/views/templates/admin/configure-bottom.tpl]: Remove empty line

File: rg_shipit/classes/ShipitCityTrack.php (hunk 1 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitCityTrack.php]: done!

File: rg_shipit/views/templates/admin/configure-bottom.tpl (hunk 2 above)
Comment (multinauta) [reply rg_shipit/views/templates/admin/configure-bottom.tpl]: done!

File: rg_shipit/classes/ShipitCore.php [hunk 3]
```diff
@@ -7,12 +7,29 @@
...
  */
 
-require(dirname(__FILE__).'/../libraries/ShipitAPI.php');
+
 require(dirname(__FILE__).'/ShipitLists.php');
 require(dirname(__FILE__).'/ShipitTools.php');
 require(dirname(__FILE__).'/ShipitCache.php');
 require(dirname(__FILE__).'/ShipitServices.php');
 require(dirname(__FILE__).'/ShipitShipment.php');
+require(dirname(__FILE__).'/ShipitCourier.php');
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitCore.php:16]: It is possible to wrap all of these classes over a folder named shipit_service and keep the same class name as implemented in Woocommerce?

File: rg_shipit/classes/ShipitCore.php [hunk 4]
```diff
@@ -126,17 +143,17 @@ protected function updateCache(Cart $cart, $hash_cart)
             if ($dest_code) {
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitCore.php]: It is possible to encapsulate all of these if statements over a function or maybe improve this logic

File: rg_shipit/classes/ShipitCourier.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
+<?php
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitCourier.php]: 2 space or 4 space?

File: rg_shipit/classes/ShipitDestiny.php [hunk 6]
```diff
@@ -0,0 +1,107 @@
+<?php
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitDestiny.php]: 2 space or 4 space?

File: rg_shipit/classes/ShipitHttpClient.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
+<?php
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitHttpClient.php]: 2 space or 4 space?

File: rg_shipit/classes/ShipitHttpClient.php [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    public $headers = array();
+    
+
+    public function __construct( $endpoint, $headers) {
+      $this->endpoint = $endpoint;
+      $this->headers = $headers;
+    }
+
+    public function get() {
+      $client = new Client();
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitHttpClient.php]: dry code here!!
create a public variable call `$client` and initialize in the constructor method and then only call the public variable

File: rg_shipit/classes/ShipitHttpClient.php [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+  class ShipitHttpClient {
+    
+    public $endpoint = '';
+    public $headers = array();
+    
+
+    public function __construct( $endpoint, $headers) {
+      $this->endpoint = $endpoint;
+      $this->headers = $headers;
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitHttpClient.php]: which `headers` you will use here?

File: rg_shipit/classes/ShipitHttpClient.php [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+use GuzzleHttp\Client;
+
+  class ShipitHttpClient {
+    
+    public $endpoint = '';
+    public $headers = array();
+    
+
+    public function __construct( $endpoint, $headers) {
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitHttpClient.php]: remove space before `$endpoint` argument definition

File: rg_shipit/classes/ShipitIntegrationCore.php [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+      );
+    }
+
+    function packages($package = array()) {
+      $client = new ShipitHttpClient($this->base.'/packages', $this->headers);
+      $response = $client->post($package);
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), package response: '.print_r($response,true));
+      } else {
+        $package = json_decode($response['body']);
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitIntegrationCore.php]: You should only return the `json_decode`.
If you return the `package` variable how you can check the result from our API?

File: rg_shipit/classes/ShipitCourier.php [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+        return $this->id;
+    }
+
+    function getSelected() {
+       return $this->selected;
+    }
+
+
+  }
+
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitCourier.php]: Close the `php` tag

File: rg_shipit/classes/ShipitDestiny.php [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    public function getDestinyId() {
+      return $this->destiny_id;
+    }
+
+    public function getName() {
+      return $this->name;
+    }
+  }
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitDestiny.php]: Close the `php` tag

File: rg_shipit/classes/ShipitGiftCard.php [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+    function getAmount() {
+      return $this->amount;
+    }
+
+    function getTotalAmount() {
+      return $this->total_amount;
+    }
+
+  }
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitGiftCard.php]: Close the `php` tag

File: rg_shipit/classes/ShipitHttpClient.php [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+        , ['json' => $body
+        , 'headers' => $this->headers]
+      );
+     
+    
+      return $response;
+    }
+    
+  }
+
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitHttpClient.php]: Close the `php` tag

File: rg_shipit/classes/ShipitSize.php [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+    public $packing_id = null;
+    public $name = '';
+
+    public function __construct($id_cart, $store = true, $packing_id = null, $name = '') {
+      $cache = new ShipitCache($id_cart);
+      $this->width = (float)$cache->package['width'];
+      $this->height = (float)$cache->package['height'];
+      $this->length = (float)$cache->package['depth'];
+      $this->weight = (float)$cache->package['weight'];
+      $this->volumetric_weight = (float)$cache->package['width']*(float)$cache->package['height']*(float)$cache->package['depth'];
```
Comment (zetahawke) [inline rg_shipit/classes/ShipitSize.php]: I see you rather prefer to take your own formula to calc volumetric weight field but, just to clarify, our couriers have their own logic to calc that field, for ex: Chilexpress is something like (width*height*depth)/4000 and others can apply the same or more.

Also I'm not aware if you are using that field at all so I will leave it as an informative warning.

* Side Note: the formula that you are applying there must give you something in orther of thousands and volumetric weight in almost any case go beyond 100

File: rg_shipit/classes/ShipitOrder.php [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+        'destiny' => $this->getDestiny(),
+        'destiny' => $this->getMongoOrderSeller()
+      );
+    }
+    
+    public function getPlatform() {
+      return $this->platform;
+    }
+
+    public function getKind(){
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitOrder.php]: Missing Empty space after parenthesis

File: rg_shipit/classes/ShipitIntegrationCore.php [hunk 18]
```diff
@@ -0,0 +1,159 @@
...
+      return $company;
+    }
+
+    function communes() {
+      $client = new ShipitHttpClient($this->base.'/communes', $this->headers);
+      $response = $client->get();
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), communes response: '.print_r($response,true));
+      } else {
+        $communes = json_decode($response->getBody());
```
Comment (zetahawke) [inline rg_shipit/classes/ShipitIntegrationCore.php]: why are you using `getBody()` and in other cases `['body']` (asking cz I'm not familiarized with this)... what I'm seeing here is the probability of reuse a lot of code and get more elasticity with the api calls.

maybe generate an abstract method that receives something like: `($method, $endpoint, $params, $entity)` you will be able to just call one method and that gives you less issues nodes at api call moment.

File: rg_shipit/classes/ShipitOrigin.php [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+    public $complement = '';
+    public $commune_id = '';
+    public $full_name = '';
+    public $email = '';
+    public $phone = '';
+    public $store = false;
+    public $origin_id = null;
+    public $name = null;
+
+    public function __construct($street = '', $number = '', $complement = '', $commune_id = '', $full_name = '', $email = '', $phone = '', $store = false, $origin_id = null, $name = null) {
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitOrigin.php]: Why in this constructor do you assign variables as a parameter if it is a constructor? they receive and assign the parameters

File: rg_shipit/classes/ShipitProduct.php [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+      public function __construct($sku_id, $amount, $warehouse_id, $description) {
+        $this->sku_id = $sku_id;
+        $this->amount = $amount;
+        $this->warehouse_id = $warehouse_id;
+        $this->description = $description;
+
+    }
+
+    function build() {
+        return array(
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitProduct.php]: Check indentation

File: rg_shipit/classes/ShipitServices.php [hunk 21]
```diff
@@ -46,10 +46,11 @@ public static function getAll($valid_id_reference = false)
      */
     public static function getByCode($code)
     {
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitServices.php]: Put in the same line as function declaration

File: rg_shipit/classes/ShipitServices.php [hunk 22]
```diff
@@ -65,11 +66,18 @@ public static function getByCode($code)
      */
     public static function getByReference($id_reference)
     {
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitServices.php]: Put in the same line as function declaration

File: rg_shipit/classes/ShipitServices.php [hunk 23]
```diff
@@ -65,11 +66,18 @@ public static function getByCode($code)
...
+        if (version_compare(_PS_VERSION_, '1.6.1.0', '>=') === true) {
+            $sql = new DbQuery();
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitServices.php]: Check indentation

**Review by zetahawke — APPROVED** [review]

File: rg_shipit/classes/ShipitShipment.php [hunk 24]
```diff
@@ -60,4 +60,13 @@ public static function getIdShipitByShipitId($shipit_id)
             WHERE shipit_id='.(int)$shipit_id
         );
     }
+
+    public static function getIdShipitByShipitOrderId($order_id)
+    {
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitShipment.php]: Put in the same line as function declaration

File: rg_shipit/classes/ShipitShipment.php [hunk 25]
```diff
@@ -60,4 +60,13 @@ public static function getIdShipitByShipitId($shipit_id)
...
+        return Db::getInstance()->getValue(
+            'SELECT `id_rg_shipit_shipment`
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitShipment.php]: Check indentation

File: rg_shipit/classes/ShipitSource.php [hunk 26]
```diff
@@ -0,0 +1,46 @@
+<?php
+  class ShipitSource {
+      public $channel;
+      public $ip;
+      public $browser;
+      public $language;
+      public $location;
+
+      public function __construct($channel = '', $ip = '', $browser = '', $language = '', $location = '') {
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitSource.php]: Sames as the other constructor

File: rg_shipit/classes/ShipitTools.php [hunk 27]
```diff
@@ -362,4 +362,37 @@ public static function setGroups($groups, $id_carrier, $delete = true)
 
         return Db::getInstance()->execute(rtrim($sql, ','));
     }
+
+
+    public static function getClientName($orderId) {
+        $shipitService = new ShipitServices();
```
Comment (mgraterol) [inline rg_shipit/classes/ShipitTools.php]: check file indentation

File: rg_shipit/classes/Shipment.php [hunk 28]
```diff
@@ -0,0 +1,71 @@
+<?php
+  class Shipment {
+      public $kind = 2;
```
Comment (mgraterol) [inline rg_shipit/classes/Shipment.php]: Check file indentation

File: rg_shipit/classes/ShipitOrigin.php (hunk 19 above)
Comment (zetahawke) [reply rg_shipit/classes/ShipitOrigin.php]: I'm not sure this works that way... If i'm correct, that are just default values that takes place on assignment when that field is not beign passed by... just default values for not getting errors like null pointer exceptions or such

File: rg_shipit/views/templates/admin/configure-bottom.tpl [hunk 29]
```diff
@@ -30,3 +37,12 @@
         </ul>
     </p>
 </div>
+
+<div class="panel">
+    <h3><i class="icon icon-tags"></i> Log de Errores</h3>
+    <p>
```
Comment (mgraterol) [inline rg_shipit/views/templates/admin/configure-bottom.tpl]: Why usin a paragraph and no a new div?

File: rg_shipit/classes/ShipitIntegrationCore.php [hunk 30]
```diff
@@ -0,0 +1,159 @@
...
+      $client = new ShipitHttpClient($this->base.'/couriers', $this->headers);
+      $response = $client->get();
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), couriers: '.print_r($response,true));
+      } else {
+        $couriers = json_decode($response->getBody());
+      }
+      return $couriers;
+    }
+
```
Comment (hiro-riveros) [inline rg_shipit/classes/ShipitIntegrationCore.php]: remove unnecessary spaces

File: rg_shipit/classes/ShipitIntegrationCore.php (hunk 18 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitIntegrationCore.php]: it should be with the getBody method, the other services I haven't used, that's why they are taking from an array, but I'll update

File: rg_shipit/classes/ShipitOrigin.php (hunk 19 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitOrigin.php]: i think it was a plugin that I used to build that class, I'll fix it (but if it works)

File: rg_shipit/views/templates/admin/configure-bottom.tpl (hunk 29 above)
Comment (zetahawke) [reply rg_shipit/views/templates/admin/configure-bottom.tpl]: p is usually used for give it a _new line_ behavior but, what you both are missing here is that list (`ul`) take the same inherit behavior so, no div and no paragraph needed if that's the case here.

Comment (zetahawke) [reply rg_shipit/views/templates/admin/configure-bottom.tpl]: though `p` is fine and I like it most than a `div` with a custom class that must give it padding and a width of 100% at least to have the same UI effect

Comment (multinauta) [reply rg_shipit/views/templates/admin/configure-bottom.tpl]: i kept the design, legacy code

File: rg_shipit/classes/ShipitOrder.php (hunk 17 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitOrder.php]: done!

**Review by mgraterol — APPROVED** [review]

**Review by hiro-riveros — CHANGES_REQUESTED** [review]

File: rg_shipit/classes/ShipitIntegrationCore.php (hunk 11 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitIntegrationCore.php]: fixed in previous amended

File: rg_shipit/classes/ShipitHttpClient.php (hunk 9 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitHttpClient.php]: Content-Type, X-Shipit-Email, X-Shipit-Access-Token and Accept

File: rg_shipit/classes/ShipitIntegrationCore.php (hunk 30 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitIntegrationCore.php]: done!

File: rg_shipit/classes/ShipitHttpClient.php (hunk 15 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitHttpClient.php]: done!

File: rg_shipit/classes/ShipitGiftCard.php (hunk 14 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitGiftCard.php]: done!

File: rg_shipit/classes/ShipitDestiny.php (hunk 13 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitDestiny.php]: done!

File: rg_shipit/classes/ShipitCourier.php (hunk 12 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitCourier.php]: done!

File: rg_shipit/classes/ShipitHttpClient.php (hunk 7 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitHttpClient.php]: done!

File: rg_shipit/classes/ShipitDestiny.php (hunk 6 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitDestiny.php]: done!

File: rg_shipit/classes/ShipitCourier.php (hunk 5 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitCourier.php]: done!

File: rg_shipit/classes/ShipitCore.php (hunk 4 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitCore.php]: done!

File: rg_shipit/classes/ShipitCore.php (hunk 3 above)
Comment (multinauta) [reply rg_shipit/classes/ShipitCore.php:16]: done!

File: rg_shipit/shipit_service/ShipitCityTrack.php [hunk 31]
```diff
@@ -0,0 +1,47 @@
+<?php
+  class ShipitCityTrack {
+    public $draft = '';
+    public $confirmed = '2019-06-07T17:13:09.141-04:00';
```
Comment (Zeaneth) [inline rg_shipit/shipit_service/ShipitCityTrack.php:4]: Just curious. What is this date about?

File: rg_shipit/shipit_service/ShipitCore.php [hunk 32]
```diff
@@ -123,50 +140,9 @@ protected function updateCache(Cart $cart, $hash_cart)
...
-                                $cache->carriers[$carrier->id] = $cost;
-                            }
-                        }
-                    }
-                }
-            }
+            $cache = $this->getCostsByCarrier($cache, $dest_code, $weight, $height, $width, $depth);
         }
-
+
```
Comment (Zeaneth) [inline rg_shipit/shipit_service/ShipitCore.php]:
```suggestion

```

File: rg_shipit/shipit_service/ShipitCore.php [hunk 33]
```diff
@@ -277,20 +253,21 @@ protected function getServicesCost($destination, $weight, $height, $width, $dept
...
             )
         );
 
         $errors = false;
-        $api = new ShipitAPI($this->config['SHIPIT_EMAIL'], $this->config['SHIPIT_TOKEN']);
-        $results = $api->getEstimateCost($params, (int)$this->config['SHIPIT_DISPATCH_ALGORITHM'], $errors);
-
+        $api = new ShipitIntegrationCore($this->config['SHIPIT_EMAIL'], $this->config['SHIPIT_TOKEN'],4);
+        $results = $api->rates($params,(int)$this->config['SHIPIT_DISPATCH_ALGORITHM']);
+
```
Comment (Zeaneth) [inline rg_shipit/shipit_service/ShipitCore.php:269]:
```suggestion

```

File: rg_shipit/shipit_service/ShipitHttpClient.php [hunk 34]
```diff
@@ -0,0 +1,52 @@
...
+    public function __construct($endpoint, $headers) {
+      $this->endpoint = $endpoint;
+      $this->headers = $headers;
+      $this->client = new Client();
+    }
+
+    public function get() {
+      $response = $this->client->get( $this->endpoint
+        , ['headers' => $this->headers
+        ,'allow_redirects'=> ['strict'=>true]
```
Comment (Zeaneth) [inline rg_shipit/shipit_service/ShipitHttpClient.php]:
```suggestion
      $response = $this->client->get(
        $this->endpoint,
        ['headers' => $this->headers,
        'allow_redirects'=> ['strict'=>true]
```
Maybe align the function options to be more readable?

File: rg_shipit/shipit_service/ShipitIntegrationOrder.php [hunk 35]
```diff
@@ -0,0 +1,64 @@
...
+        $order = json_decode($response->getBody());
+      }
+      return $order->id;
+    }
+
+    function massiveOrders($order = array()) {
+      $client = new ShipitHttpClient($this->base.'/orders/massive', $this->headers);
+      $response = $client->post($order);
+      if ($response->getStatusCode() != 200) {
+        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), orsers massive response: '.print_r($response,true));
```
Comment (Zeaneth) [inline rg_shipit/shipit_service/ShipitIntegrationOrder.php]:
```suggestion
        ShipitTools::log('PrestaShop ('._PS_VERSION_.'), orders massive response: '.print_r($response,true));
```

File: rg_shipit/shipit_service/ShipitServices.php [hunk 36]
```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Integral logistics for eCommerce with pickup or fulfillment through Shipit
+ *
+ * @author    Rolige <www.rolige.com>
+ * @copyright 2011-2018 Rolige - All Rights Reserved
+ * @license   Proprietary and confidential
+ */
```
Comment (Zeaneth) [inline rg_shipit/shipit_service/ShipitServices.php]: Is this necessary?

**Review by Zeaneth — APPROVED** [review]
I've seen it working and I just suggested some changes to the code order.

File: rg_shipit/shipit_service/ShipitServices.php (hunk 36 above)
Comment (multinauta) [reply rg_shipit/shipit_service/ShipitServices.php]: no, i will remove it

File: rg_shipit/shipit_service/ShipitCityTrack.php (hunk 31 above)
Comment (multinauta) [reply rg_shipit/shipit_service/ShipitCityTrack.php:4]: in the original model it was like this, I kept it, but it was one of the two classes that is no longer used but remains

File: rg_shipit/shipit_service/ShipitCore.php (hunk 32 above)
Comment (multinauta) [reply rg_shipit/shipit_service/ShipitCore.php]: done!

File: rg_shipit/shipit_service/ShipitCore.php (hunk 33 above)
Comment (multinauta) [reply rg_shipit/shipit_service/ShipitCore.php:269]: done!

File: rg_shipit/shipit_service/ShipitHttpClient.php (hunk 34 above)
Comment (multinauta) [reply rg_shipit/shipit_service/ShipitHttpClient.php]: done!

File: rg_shipit/shipit_service/ShipitIntegrationOrder.php (hunk 35 above)
Comment (multinauta) [reply rg_shipit/shipit_service/ShipitIntegrationOrder.php]: done!

**Pull request merged**